### ISSUE TITLE
Update `BuyersGuideCallToAction`'s  "link_url" field to CharField

### ISSUE THREAD

**Issue by danielfmiranda** [MozillaFoundation/foundation.mozilla.org#10514]
## Description

At the moment, the `BuyersGuideCallToAction`'s `link_target_url` field is set to use Django's [`URLfield`](https://docs.djangoproject.com/en/4.2/ref/models/fields/#urlfield). 

This is useful if we want the CTA to link to another URL, such as "donate.mozilla.org".

However, since we have now updated the foundation site to render the donate modal whenever the UTM param `?form=donate` is appended to the current URL, it would be nice to have the CTA be able to append it as well.

Unfortunately, setting the field to `?form=donate` does not pass the URL fields validation, as it expects a full URL with `http://` included.

We can achieve this however by updating the `URLField` to `CharField` instead, and preforming the validation within the `clean()` method using a regex expression.

**We would need to make sure the attempted value falls under one of these 4 categories:**

1. The string is a valid URL. Example: `http://google.com/`
2. The string is a valid URL with query parameters. Example: `http://google.com/?form=donate`
3. The string is a valid query parameter. Example: `?form=donate`
4. The string is a valid group of query parameters. Example: `?form=donate&campaign_year=2022`

A regex expression that I found from a quick google search is:
```
^(http|https):\/\/[a-z0-9\-\.]+\.[a-z]{2,}(\/\S*)?(\?.+)?$|^(\?.+)$|^(\?.+)(\&.+=.+)*$
```

and the validation seems to work as expected.

## Dev Tasks:

- [ ] Update `link_target_url` from a `URLField` to a `CharField`, we are going to want to make sure that any existing data for this field gets carried over! 
- [ ] Add validation for this fields value in the models `clean()` method. An example of this being done can be found [here](https://github.com/MozillaFoundation/foundation.mozilla.org/blob/2cd60a5f17913fb058cf8759bc494f507df5de46/network-api/networkapi/wagtailpages/pagemodels/blog/blog.py#L325). (**Note:** the example is preforming the validation within the `clean()` method directly, however, I would create a helper function to do the validation, and then call that within the clean method instead.
- [ ] Add some tests for the validation 

## Acceptance Criteria

- [ ] As a CMS user, I can set the `link_target_url` value to be any of the following formats:
  - Valid URL. Example: `http://google.com/`
  - Valid URL with query parameters. Example: `http://google.com/?form=donate`
  - Valid query parameter. Example: `?form=donate`
  - Valid group of query parameters. Example: `?form=donate&campaign_year=2022`
- [ ] As a CMS user, if I set the `link_target_url` value to anything else besides the 4 formats above, a descriptive validation error is returned.

**Comment by kristinashu** [comment]
Yes, this is working great! The FRU modal will now open on the page a user is on! So much better.

It is unclear to me if the utm is getting carried through tho? The snippet url is ?form=donate&utm_medium=FMO&utm_source=PNI&utm_campaign=23-PNI&utm_content=banner&c_id=7014x000000eQOD so on https://foundation.mofostaging.net/en/privacynotincluded/categories/mental-health-apps/ the link becomes https://foundation.mofostaging.net/en/privacynotincluded/categories/mental-health-apps/?form=donate&utm_medium=FMO&utm_source=PNI&utm_campaign=23-PNI&utm_content=banner&c_id=7014x000000eQOD but I don't see that url once the modal has opened. Is there a way to test if it is going through?

**Comment by danielfmiranda** [comment]
Hi @kristinashu I have taken a look on staging and can see what you mean, it appears that even though the relative link includes all of the UTM parameters, something is happening on the fundraise up side that is removing them from the URL when rendering the form.

However, after making a test donation it appears that the UTM parameters are still being tracked on the Google Tag Manager side! 

![Screenshot 2023-05-16 at 11-13-42 Tag Assistant Connected.png](https://images.zenhubusercontent.com/6046b7f1bc36adc9ce95a16a/711e72cd-f111-4924-89ef-07426f959c49)

**Comment by kristinashu** [comment]
That's great, thank you for checking! I'm guessing that means the utm is also carried over to Salesforce?

**Comment by danielfmiranda** [comment]
Hi @kristinashu! I believe so, however since I don't have access to Salesforce I am not able to test

**Comment by kristinashu** [comment]
Ok, Kevin can test once it's on prod!
